presto-main/src/main/java/com/facebook/presto/sql/gen/InputReferenceCompiler.java

```diff
@@ -106,4 +95,61 @@ public BytecodeNode visitVariableReference(VariableReferenceExpression reference
     {
         throw new UnsupportedOperationException();
     }
+
+    static class InputReferenceNode
+            implements BytecodeNode
+    {
+        IfStatement body;
```

Comment:
nit: Make type to be `BytecodeNode`. 

Make these three fields `private final`